crates/wasmtime/src/store.rs

```diff
@@ -1052,10 +1088,33 @@ impl<T> StoreInner<T> {
     }
 
     pub fn call_hook(&mut self, s: CallHook) -> Result<(), Trap> {
-        if let Some(hook) = &mut self.call_hook {
-            hook(&mut self.data, s)
+        // Need to borrow async_cx before the mut borrow of the limiter.
+        // self.async_cx() panicks when used with a non-async store, so
+        // wrap this in an option.
```

Comment:
Is this still applicable? It looks like `self.inner.async_cx()` is disjoint from `self.call_hook` which is part of the `match` so I think this should be able to move in to the `Async` block below?

Reply:
Yes, seems right. Also, moving it removes an unnecessary test if the `async` feature is set but we're not actually using `async` callbacks.

This code was mirrored from `memory_growing`, so I've updated that function, too.

Addressed in fdbd3b9.